### DISCUSSION TITLE
Allow (optional) prev/next change navigation within visible page in Text Diff

### DISCUSSION THREAD

**Discussion by goran-w** [sourcegit-scm/sourcegit#750] [Ideas]
Enhancement/suggestion #616 was closed after implementing a version which is different from my suggested implementation in PR #703. This was done with a reasoning that change navigation within the currently visible "page" is unnecessary (which is where I beg to differ).

The currently implemented feature (released in [v8.39](https://github.com/sourcegit-scm/sourcegit/releases/tag/v8.39)) is basically an enhanced Page Up/Down, which jumps to the previous/next "page" that contains a change not already shown on the current "page". No highlighting is made, since there is no notion of a "current" change.

In contrast, my PR adds the following features which are more similar to how P4Merge, WinMerge and other external Diff/Merge tools do their change navigation:

* Jump (scroll) to prev/next change-block (defined as a consecutive block of added/removed lines).
* Highlight the "current change-block".
* Show numbers for "current change-block" and "total number of change-blocks" in the Diff toolbar. (Example: `3/15`)
* For each new Diff, the "current change-block" is initially unset (Example:` -/15`) and no highlighting is done until a prev/next arrow is clicked.
* Prev/next arrows are not disabled when at first/last change-block, instead clicking the arrow in these cases will scroll the first/last change-block into view again.

I would like this alternative implementation (from the PR) to be included as an optional feature, selectable by an option in Preferences. The current implementation could still be the default for this option. (See ongoing discussion in the PR.)

NOTE: Keyboard shortcuts could be added as a future enhancement.

**Comment by goran-w** [comment]
Here's a screenshot from the implementation in my PR #703, showing the "current/total" indicator and a highlighted "current" change-block :

![image](https://github.com/user-attachments/assets/7d48d24d-e31a-4d8e-95f8-38660d21c6e5)

**Comment by goran-w** [comment]
Added button "Highlighted Diff Navigation" for toggling the two navigation styles, using a "highlighter pen" icon.
![image](https://github.com/user-attachments/assets/2b3d00ad-3546-4ac8-9d33-07a12e058775)

Feature enabled (ToggleButton ON, additional current/total numbers visible between the Prev/Next difference buttons, current change-block highlighted in editor panels) :
![image](https://github.com/user-attachments/assets/c23e3ba7-ad76-4f05-ba9e-d1f12f71499e)

Feature disabled (default behavior, ToggleButton OFF, no number indicator, no highlighting) :
![image](https://github.com/user-attachments/assets/0d2dda44-5374-4f1e-bd05-0cfa01fd4ca9)

**Comment by goran-w** [comment]
PR #703 is ready for merge, after rework to make it optional (via ToggleButton and Preference variable).

**Comment by goran-w** [comment]
@love-linger Would you mind merging PR #703? It's been kept up to date (rebased) with the Develop branch and can now be merged without affecting the default behavior, since my implementation is now OFF by default and can be toggled ON from (Text) DiffView toolbar.

**Comment by love-linger** [comment]
I've merged this PR, and add some modifications. You can checkout the latest `develop` branch to see the details.
